plugins/modules/packaging/os/snap.py

```diff
@@ -171,6 +212,45 @@ def _run_multiple_commands(self, commands):
             '\n'.join(results[3]),
         ]
 
+    def convert_json_subtree_to_map(self, json_subtree, prefix=None):
+        option_map = {}
+
+        for key, value in json_subtree.items():
+            full_key = key if prefix is None else prefix + "." + key
+
+            if isinstance(value, str) or isinstance(value, int) or isinstance(value, float) or isinstance(value, bool):
+                option_map[full_key] = str(value)
+
+            else:
+                option_map.update(self.convert_json_subtree_to_map(json_subtree=value, prefix=full_key))
```

Comment:
What happens if a list is present? Right now it is handled here, and will cause a crash since it does not have `.items()`.

Also: I would probably rearrange the code to check `isinstance(value, dict)` first and if that's `True` do this one, and if not convert the value to string. That is less brittle.

(One case you didn't handle is `long` on Python 2 for example. If someone passes a long integer and this module runs on Python 2, the code will try to call `value.items()` and will crash.)

Reply:
The new code explicitly checks for a dict element now. Lists should not be returned in the option map and will trigger an exception now. As for the long values of Python2 there's a new check for numbers.Integral which covers both int and long.